src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,119 @@
+        return input.equals("list");
+    }
+
+    public static boolean isBye() {
+        return input.equals("bye");
+    }
+
+    public static boolean isDone() {
+        return input.startsWith("done");
+    }
+
+    public static boolean isEvent() {
+        return input.startsWith("event");
+    }
+
+    public static boolean isDeadLine() {
+        return input.startsWith("deadline");
+    }
+
+    public static boolean isToDo() {
+        return input.startsWith("todo");
+    }
+
+    public static boolean isDelete() { return input.startsWith("delete"); }
+
+    //check if there is input exception
+    public static boolean isInvalidToDo() {
+        return input.substring(TODO_LENGTH).equals("");
+    }
+
+    public static boolean isInvalidDeadLine() {
+        String inputTemp = input.substring(DEADLINE_LENGTH).trim();
+        boolean isEmpty = inputTemp.equals("");
+        String[] task = inputTemp.split("/by");
+        return (isEmpty || task.length == 0 || task.length == 1);
+
+    }
+
+    public static boolean isInvalidEvent() {
+        String inputTemp = input.substring(EVENT_LENGTH).trim();
+        boolean isEmpty = inputTemp.equals("");
+        String[] task = inputTemp.split("/at");
+        return (isEmpty || task.length == 0 || task.length == 1);
+    }
+
+
+    /**
+     * Convert the string that user inputs to a specific task object
+     * Categorize them into todos, deadlines, and events
+     */
+    public static Task convertInputToTask()
+            throws InvalidDeadlineTimeException, InvalidEventTimeException, EmptyToDoException, InvalidCommandException {
+        if(isDeadLine()){
+            if(isInvalidDeadLine()) {
+                throw new InvalidDeadlineTimeException();
+            }
+            return new DeadLine(input);
+        }
+        if(isEvent()){
+            if(isInvalidEvent()) {
+                throw new InvalidEventTimeException();
+            }
+            return new Event(input);
+        }
+        if(isToDo()){
+            if(isInvalidToDo()) {
+                throw new EmptyToDoException();
+            }
+            return new ToDo(input);
+        }
+        else
+            throw new InvalidCommandException();
+    }
+
+
+    public static void runCommand(String input, Command c)
+            throws TaskIndexOutOfBound, InvalidDeadlineTimeException, InvalidEventTimeException, EmptyToDoException, InvalidCommandException, EmptyDoneIndexException, InvalidInputException{
+        switch (c) {
+        case LIST:
+            taskList.listTasks();
+            break;
+        case DONE:
+            taskList.tasksDone(input);
+            break;
+        case DELETE:
+            taskList.deleteTask(input);
+            break;
+        case INVALID:
+            throw new InvalidInputException();
+        case TODO:
+        case EVENT:
+        case DEADLINE:
+            Task task = convertInputToTask();
+            taskList.addTask(task);
+            break;
```

Comment:
Perhaps handling the cases with different methods instead of fall through cases like these would be better, to maintain the [logical structure of the code](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#structure-code-logically). 

Arguably the input conversion may be the same but you could factor that in elsewhere.